### DISCUSSION TITLE
[Feature] Stacks working in albums

### DISCUSSION THREAD

**Discussion by lordkitsuna** [immich-app/immich#6387] [Feature Request]
### The feature

As of right now stacking only works in the main timeline, extend the feature to function when a stack is sent to albums as well

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by maltechx** [comment]
I would also like to see this. I have external libraries in Immich that contain images with raw and jpeg versions. In the timeline I stack these images, when I add this stack I have two images in the album.

**Comment by mirisbowring** [comment]
Another UseCase:

I shoot RAW+JPG with my camera and stack them in my timeline. After, I add them to an album to share the trip with my friends. Now, every image is shown twice (one times a bit darker, one a bit brighter - due to beeing raw or jpeg).
Most of them are irritated why they are duplicated - But some definetily want the RAWS which is why i can't just share the JPEGs

**Comment by Gecko-with-a-hat** [comment]
Are there any plans to implement this feature as of yet? Personally I have postcards that I digitize to store them forever. To then have the front, inside and back together as one I stack them on the timeline. This makes it a really neat way to have almost like a digital way to open and close the postcard. Love to also have this feature in albums so that I can share the album and have it load up nicely.

**Comment by rswafford** [comment]
I would like to see this working in albums as well, especially shared albums. I am scanning old photos with writing on the back, and stacking the front and back scans together to keep the metadata together. Any eta on this working in albums?

**Comment by Nikishek** [comment]
When I stack photos on the timeline and then share the stacked "photo" to an album, only one photo from the stack is in the shared album. This is not good.

Similarly, if I already have a shared album full of photos and I stack those photos on the timeline, it won't show up in the shared album. This is a similar "issue".

If I want to do this again time in the album, this stacking function is not available.

**Reply by kodidu** [reply]
I am a first time user and exactly suffered from all the stacking issues that you described.  

Yes, the stacking feature would/is great.  
But at the current state it is useless, because it is solely tied to the timeline. Stacking should maybe be tied to the photos.

**Comment by haoenz** [comment]
I am really looking forward to this feature! I think that compared to the timeline, the album should only have additional functions, not less functions.

**Comment by connorearl** [comment]
Also running into this issue when sharing albums with friends. Just stacked everything with https://github.com/tenekev/immich-auto-stack and then realized it doesn't show up in albums anyway.

**Comment by MagnusL3D-code** [comment]
+1 from me, very much need stacks in albums.

**Comment by ds-281** [comment]
Another use case: I have an album called "woodworking projects" and I'd like to stack the pictures on a project basis (I have a few projects in parallel and timestamp-based ordering isn't ideal, stacks are a super nice way to solve this).

**Comment by LnLcFlx** [comment]
I think this feature would complement the "auto stacking" on the roadmap very well.

It is important though, to consider that some people might want the possibility to add a stacked image without adding all other images from this stack.
I personally don't need this but I can imagine the following scenario:
- I usually put edited/censored images (like pictures with license plate that I want to publish) in one stack with the original picture.
- If I want to share such pictures in a shared album I only want the censored picture, not the original.

To deal with this in a simple and powerful way, I would introduce a global "un-/show stacks" button that switches between showing all pictures separately and showing them in stacks.
Depending on which mode you are in, a single picture, or the entire stack gets added to an album.
I think this decision should also be remembered in the album, i.e. truly adding a stack and not just all pictures that are in the stack right now, because of the following question:

> Should two separately added pictures from one stack (added from the "unstacked view") also be displayed saparately in the album or as part of one stack?

I think yes, and for pictures added to an album separately from their stack, we need a drop down "add the entire stack of this picture".
This allows for a clean handling of the following...

> When adding a stack to an album and then adding a picture to this stack, should this new image automatically appear in the album too?

I think yes, if the whole stack has been added to the album before and no if only a singular picture from the stack was added.

This all might sound a bit cluttered, but with the right UI, it is rather intuitive.

**Comment by belst** [comment]
I just wanted to share a bunch of stacked photos via an album, but it unstacks them. really unintuitive and unexpected behaviour

**Comment by ThomasConrad** [comment]
Can we please have this, all my albums get so cluttered 😭

**Comment by bo0tzz** [comment]
Closing in favour of #16549

**Comment by pbaniewicz** [comment]
Another vote for stacking in albums I have a ton of fronts and backs which look  terrible in albums

**Comment by softyoda** [comment]
Same "issue" the stack is only visible in the main view, not in album view, it create lot of recto and verso visibe at the same time
